File: Grocery-Backend/frontend/src/admin/services/productService.js
import axiosInstance from './axiosInstance';

export const productService = {
  getProducts: async (page = 0, size = 8, sortBy = 'productId', direction = 'desc') => {
    const res = await axiosInstance.get(`/products?page=${page}&size=${size}&sortBy=${sortBy}&direction=${direction}`);
    return res.data;
  },
  
  searchProducts: async (keyword) => {
    const res = await axiosInstance.get(`/products/search?keyword=${encodeURIComponent(keyword)}`);
    return res.data;
  },

  getProductsByCategory: async (category) => {
    const res = await axiosInstance.get(`/products/category/${category}`);
    return res.data;
  },

  getProductById: async (productId) => {
    const res = await axiosInstance.get(`/products/${productId}`);
    return res.data;
  },

  addProduct: async (productData) => {
    const res = await axiosInstance.post('/products', productData);
    return res.data;
  },

  updateProduct: async (productId, productData) => {
    const res = await axiosInstance.put(`/products/${productId}`, productData);
    return res.data;
  },

  deleteProduct: async (productId) => {
    const res = await axiosInstance.delete(`/products/${productId}`);
    return res.data;
  }
};

File: Grocery-Backend/src/main/java/com/grocery/service/ProductService.java
package com.grocery.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.grocery.dto.ProductPageResponse;
import com.grocery.dto.ProductRequest;
import com.grocery.dto.ProductResponse;
import com.grocery.dto.ProductUpdateRequest;
import com.grocery.enums.Category;

public interface ProductService {

	  ProductResponse addProduct(ProductRequest request);

	    ProductResponse getProductById(Long productId);

//	    List<ProductResponse> getAllProducts();

	    ProductResponse updateProduct(Long productId,
	                                  ProductUpdateRequest request);

	    void deleteProduct(Long productId);
	    
	    List<ProductResponse> searchProducts(String keyword);
	    
	    List<ProductResponse> getProductsByCategory(Category category);

	    ProductPageResponse getAllProducts(
	            int page,
	            int size,
	            String sortBy,
	            String direction);

	    // Products added by the currently logged-in supplier
	    List<ProductResponse> getMyProducts();
	    
	  
	    
}
